mcstatus/motd.py

```diff
@@ -0,0 +1,280 @@
+
+        return raw_motd
+
+    @classmethod
+    def _parse_as_str(cls, raw: str, *, bedrock=False) -> Self:
+        """Parse a MOTD when it's string.
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition?
+            Ignores ``MinecraftColor.MINECOIN_GOLD`` if it's ``False``.
+        :returns: New ``MOTD`` instance.
+        """
+        raw_motd: RawMotd = []
+
+        splited_raw: List[str] = MOTD_COLORS_RE.split(raw)
+        for element in splited_raw:
+            clean_element = element.lstrip("&§")
+            if clean_element == "g" and not bedrock:
+                raw_motd.append(element)  # minecoin_gold on java server
+                continue
+            try:
+                raw_motd.append(MinecraftColor(clean_element))
+            except ValueError:
+                try:
+                    raw_motd.append(Formatting(clean_element))
+                except ValueError:
+                    # just a text
+                    raw_motd.append(element)
+
+        # `for` loop for removing empty values
+        return cls([e for e in raw_motd if e], bedrock)
+
+    @property
+    def plain(self) -> str:
+        """Getter for string variant of a MOTD.
+
+        :returns: String variant of MOTD.
+
+        .. note:: This will always use "§", even if in original MOTD everywhere was "&".
+        """
+        result = ""
+        for element in self.raw:
+            if isinstance(element, str):
+                result += element
+                continue
+            if isinstance(element, WebColor):
+                continue
+
+            result += "§" + element.value
+
+        return result
+
+    @property
+    def html(self) -> str:
+        """Getter for HTML variant of a MOTD.
+
+        .. warn::
+            You should implement obfuscated CSS class yourself (name - ``obfuscated``).
+            See `this answer <https://stackoverflow.com/a/30313558>`_ as example.
+
+        :returns: String that contain HTML.
+        """
+        result = "<p>"
+        on_reset = ""
+        for element in self.raw:
+            if isinstance(element, str):
+                result += element
+            elif isinstance(element, WebColor):
+                result += f"<span style='color:{element.hex}'>"
+                on_reset += "</span>"
+
+            elif isinstance(element, Formatting):
+                if element == Formatting.RESET:
+                    result += on_reset
+                    on_reset = ""
+                elif element == Formatting.OBFUSCATED:
+                    result += "<span class=obfuscated>"
+                    on_reset += "</span>"
+                else:
+                    tag_name = {
+                        Formatting.BOLD: "b",
+                        Formatting.STRIKETHROUGH: "s",
+                        Formatting.ITALIC: "i",
+                        Formatting.UNDERLINED: "u",
+                    }[element]
```

Comment:
This is just a dict, I understand what wrong was with overloads in #306, but here - just a dict.

Reply:
? I haven't mentioned anything about overloads, all I said is that these dicts should be global variables, and they shouldn't be defined here in the function body, because that means they'll keep getting re-created on each function call, which is pointless.

Reply:
> I haven't mentioned anything about overloads

I mean, [this conversation](https://github.com/py-mine/mcstatus/pull/306#discussion_r872961673) from #306.

Reply:
Changed [here](https://github.com/py-mine/mcstatus/compare/bccee991168674292f251d33cc2c2f19584abfb9..b09a9e2fd06e0109fd3a25b7357725f1e80d9034).